### PR TITLE
Fixed batchnorm bug

### PR DESCRIPTION
**Pull request by cehongwang**
# Description

Batch norm value trace fails when the intial weights are identical. This pull request fixed batch norm fast refit case using name trace.

Fixes # (issue)

## Type of change

Please delete options that are not relevant and/or add your own.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

# Checklist:

- [ ] My code follows the style guidelines of this project (You can use the linters)
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests to verify my fix or my feature
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

**Review by narendasan — COMMENTED** [review]
@zewenli98 what are your thoughts here, seems like a simple enough change. @cehongwang after this is in would there be any other failure modes for fast refit?

File: py/torch_tensorrt/dynamo/conversion/impl/normalization/ops.py [hunk 1]
```diff
@@ -50,14 +50,27 @@ def batch_norm(
...
-    if running_var is None:
-        running_var = get_trt_tensor(ctx, 1.0, f"{name}_running_var")
+    # We name the weight here according to the state_dict name
+    weight = (
+        get_trt_tensor(ctx, 1.0, f"{name}_weight")
+        if weight is None
+        else get_trt_tensor(ctx, weight, f"{name}_weight")
+    )
+    bias = (
+        get_trt_tensor(ctx, 1.0, f"{name}_bias")
```
Comment (zewenli98) [inline py/torch_tensorrt/dynamo/conversion/impl/normalization/ops.py:60]: Should this be 0?

File: py/torch_tensorrt/dynamo/conversion/impl/normalization/ops.py [hunk 2]
```diff
@@ -50,14 +50,27 @@ def batch_norm(
...
+        if bias is None
+        else get_trt_tensor(ctx, bias, f"{name}_bias")
+    )
+    running_mean = (
+        get_trt_tensor(ctx, 1.0, f"{name}_running_mean")
```
Comment (zewenli98) [inline py/torch_tensorrt/dynamo/conversion/impl/normalization/ops.py:65]: Should this be 0?

**Review by zewenli98 — COMMENTED** [review]
Is this PR still a draft or ready for review (rebase to main is needed)?

**Comment by zewenli98** [comment]
> @zewenli98 what are your thoughts here, seems like a simple enough change. @cehongwang after this is in would there be any other failure modes for fast refit?

Since batchnorm was refactored previously, renaming maybe work for fast refit.

**Comment by cehongwang** [comment]
I am setting up the naming trace for fast refit by name the trt weight same
as what is in the state dict. This is still a draft I am having trouble
with installing the code lint on my current computer. The code is running
and passed the tests though. Will convert to a PR after fixing the lint.

**Comment by cehongwang** [comment]
> @zewenli98 what are your thoughts here, seems like a simple enough change. @cehongwang after this is in would there be any other failure modes for fast refit? 

I will do similar changes to other applicable converters to make the naming trace more robust. There should not be other failures after that because we have the value trace to guarantee the correctness.

**Comment by zewenli98** [comment]
@cehongwang @narendasan Can you guys take a look at the issue #3200 which may be related to this PR

**Review by zewenli98 — APPROVED** [review]
LGTM

File: examples/dynamo/refit_engine_example.py [hunk 3]
```diff
@@ -69,7 +69,8 @@
     debug=debug,
     min_block_size=min_block_size,
     torch_executed_ops=torch_executed_ops,
-    make_refittable=True,
+    make_refitable=True,
```
Comment (narendasan) [inline examples/dynamo/refit_engine_example.py:72]: Should be 2 ts now, might want to rebase this branch

File: py/torch_tensorrt/dynamo/conversion/_TRTInterpreter.py [hunk 4]
```diff
@@ -477,12 +477,18 @@ def _save_weight_mapping(self) -> None:
                     # Retrieve each weight name(s) in state_dict
                     if layer_type == "CONSTANT":
                         if "embedding" in suffix:
-                            sd_weight_name = f"{sd_weight_name}.{torch_attr[0]}"
```
Comment (narendasan) [inline py/torch_tensorrt/dynamo/conversion/_TRTInterpreter.py:480]: @zewenli98 keep track of this, seems like there could be a lot of possible names we need to have might want to look at a generic solution later

**Review by narendasan — COMMENTED** [review]
Rebase and lgtm

File: examples/dynamo/refit_engine_example.py (hunk 3 above)
Comment (zewenli98) [reply examples/dynamo/refit_engine_example.py]: rebased

**Review by narendasan — APPROVED** [review]
LGTM!
